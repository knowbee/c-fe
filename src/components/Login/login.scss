.right-side {
    height: 100vh;

  .login-form {
    border-radius: 10px;
    background-color: #fff;
    width: 50%;
    margin: 5rem auto;
    padding: 2rem;
    box-shadow: 1px -5px 10px 5px rgba(0, 0, 0, 0.116);
    .error{
      text-align: center;
      color: red;
    }
    .login-form-header {
      border-bottom: 1px solid #ddd;
      margin-left: -2.1rem;
      margin-right: -2.1rem;
      margin-bottom: 3rem;

      ul {
        margin: 0rem;
        padding: 0rem;

        li {
          list-style: none;
          border-bottom: 2px solid #2aa2fa;
          width: 5.3rem;
          padding-bottom: 0.5rem;
          margin: auto;
        }
      }
    }

    form {
      display: flex;
      flex-direction: column;

      input {
        margin-bottom: 1.2rem;
        padding: 0.8rem;
        border: 1px solid #ddd;
      }

      .terms-conditions {
        display: flex;
        flex-direction: row;

        p {
          margin-top: 0rem;
          margin-left: 0.5rem;

          span {
            color: #2aa2fa;
          }
        }
      }

      .login-button {
        background-color: #2aa2fa;
        color: #fff;
        box-shadow: none;
        border-radius: 5px;
        border: none;
        padding: 0.5rem;
        cursor: pointer;
        margin-top: 2rem;
        margin-bottom: 1rem;
        font-size: 0.9rem;
      }

      .register-button {
        background-color: #fff;
        color: #2aa2fa;
        box-shadow: none;
        border-radius: 5px;
        border: none;
        padding: 0.5rem;
        cursor: pointer;
        font-size: 0.9rem;
      }
    }
  }
}

@media (max-width: 999px) {

  .layout{

    .right-side,.left-side{
      width: 100%;
      display: flex;
      align-items: center;
      justify-content: center;
      .login-form{
        width: 70%;
      }
    }
  }

}