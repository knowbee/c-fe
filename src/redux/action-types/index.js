export * from "./registerUserTypes";
export * from "./loginTypes";
export * from "./currentUserTypes";
export * from "./usersTypes";
