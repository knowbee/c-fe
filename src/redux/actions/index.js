export * from "./registerActions";
export * from "./loginActions";
export * from "./currentUserActions";
