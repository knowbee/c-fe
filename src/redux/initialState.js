module.exports = {
  currentUser: {
    message: "",
    isLoggedIn: localStorage.getItem("token") !== null,
    profile: {},
  },
  register: {
    submitting: false,
    message: "",
    email: "",
    name: "",
    password: "",
    errors: {},
  },
  users: {
    fetching: false,
    chats: [],
    errors: {},
  },
  login: {
    submitting: false,
    credentials: {
      email: "",
      password: "",
    },
  },
};
