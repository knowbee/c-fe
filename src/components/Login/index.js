import React, { useState } from "react";
import "./login.scss";
import Layout from "../Layout";
import { useHistory } from "react-router-dom";

function Login() {
  const history = useHistory();

  const goToRegister = () => {
    history.push("/");
  };

  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");

  return (
    <Layout>
      <div className="login-form">
        <div className="login-form-header">
          <img src="https://iili.io/fi0ROQ.png" alt="logo-2" />
          <ul>
            <li>Login</li>
          </ul>
        </div>
        <form>
          <input
            value={email}
            onChange={(e) => setEmail(e.target.value)}
            type="text"
            placeholder="E-mail"
          ></input>
          <input
            value={password}
            onChange={(e) => setPassword(e.target.value)}
            type="text"
            placeholder="Password"
          ></input>

          <button className="login-button" type="submit">
            Login
          </button>
          <button
            onClick={goToRegister}
            className="register-button"
            type="submit"
          >
            Create account
          </button>
        </form>
      </div>
    </Layout>
  );
}

export default Login;
