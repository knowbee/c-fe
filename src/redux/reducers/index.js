import { combineReducers } from "redux";
import register from "./registerReducer";
import login from "./loginReducer";
import currentUser from "./currentUserReducer";
import users from "./usersReducer";

const reducer = combineReducers({
  register,
  login,
  currentUser,
  users,
});
export default reducer;
