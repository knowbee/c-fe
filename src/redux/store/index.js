import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import { composeWithDevTools } from "redux-devtools-extension/developmentOnly";
import initialState from "../initialState";
import reducers from "../reducers/index";

const store = createStore(
  reducers,
  initialState,
  composeWithDevTools(applyMiddleware(thunk))
);

export default store;
