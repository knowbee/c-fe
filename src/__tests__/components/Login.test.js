import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { createStore } from "redux";

import "./index.css";
import rootReducer from "../../redux/reducers";
import Login from "../../components/Login";

const store = createStore(rootReducer);

test("Login page loads with default state", () => {
  const component = renderer.create(
    <Provider store={store}>
      <Login />
    </Provider>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
