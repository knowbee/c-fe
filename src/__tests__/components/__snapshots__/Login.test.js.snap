// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page loads with default state 1`] = `
<div
  className="layout"
>
  <div
    className="left-side"
  >
    <div
      className="middle-container"
    >
      <img
        alt="logo"
        src="https://iili.io/fi0IUb.png"
      />
      <p>
        stay in touch with your friends anytime and anywhere
      </p>
    </div>
  </div>
  <div
    className="right-side"
  >
    <div
      className="login-form"
    >
      <div
        className="login-form-header"
      >
        <img
          alt="logo-2"
          src="https://iili.io/fi0ROQ.png"
        />
        <ul>
          <li>
            Login
          </li>
        </ul>
      </div>
      <form>
        <input
          onChange={[Function]}
          placeholder="E-mail"
          type="text"
          value=""
        />
        <input
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
        <button
          className="login-button"
          onClick={[Function]}
          type="submit"
        >
          Login
        </button>
        <button
          className="register-button"
          onClick={[Function]}
          type="submit"
        >
          Create account
        </button>
      </form>
    </div>
  </div>
</div>
`;
